qemu/tests/migration.py

```diff
@@ -219,10 +214,14 @@ def guest_stress_deamon():
             target_mig_parameters = params.get("target_migrate_parameters", "None")
             target_mig_parameters = ast.literal_eval(target_mig_parameters)
             migrate_parameters = (mig_parameters, target_mig_parameters)
+            pre_migrate = get_functions(params.get("pre_migrate"), globals())
 
             # Migrate the VM
             ping_pong = params.get("ping_pong", 1)
             for i in range(int(ping_pong)):
+                # run some functions before migrate start
+                for func in pre_migrate:
+                    func(vm, params, test)
```

Comment:
Logically speaking this change looks good to me. I've only two questions:

1. Are those pre_migrate functions all idempotent? Otherwise we would need to split in, e.g., pre_migrate_once and pre_migrate.

2. Would make sense to run the post_migrate functions at end of each iteration as well?

Reply:
(1) I actually mentioned it in the commit message, those that I know of should be but there might be some downstream implementations I'm not aware of.

(2) I'm not using any `post_migrate` functions so I can't really tell (don't see usages in `migrate.cfg` either). So I'd rather wait for some user to fix that, should it require fixing..

Reply:
Ok @ldoktor , thanks for your reply.